src/NoUnused/Exports.elm

```diff
@@ -547,6 +553,46 @@ fromProjectToModule =
         |> Rule.withModuleDocumentation
 
 
+declarationToTopLevelExpose : Range -> Declaration -> Maybe TopLevelExpose
+declarationToTopLevelExpose range declaration =
+    case declaration of
+        Declaration.FunctionDeclaration function ->
+            function.declaration
+                |> Node.value
+                |> .name
+                |> Node.value
+                |> Exposing.FunctionExpose
+                |> Just
+
+        Declaration.AliasDeclaration typeAlias ->
+            typeAlias.name
+                |> Node.value
+                |> Exposing.TypeOrAliasExpose
+                |> Just
+
+        Declaration.CustomTypeDeclaration type_ ->
+            Just <|
+                Exposing.TypeExpose
+                    { name = Node.value type_.name
+                    , open = Just range
+                    }
+
+        Declaration.PortDeclaration signature ->
+            signature.name
+                |> Node.value
+                |> Exposing.FunctionExpose
+                |> Just
+
+        Declaration.InfixDeclaration infix ->
```

Comment:
I'm not sure how to test this. I thought Elm didn't have custom infix operators anymore.

Reply:
`elm-syntax` aims to be able to support code from core packages, hence why this is a possibility, but it's not one we have to consider so this is fine :+1: